components/apply/RoleSpecificQuestions.tsx

```diff
@@ -105,8 +102,14 @@ const RoleSpecificQuestions: FC<Props> = ({
           ?.filter((question): question is AggregatedQuestion => !!question),
       }))
       .filter(({ questions }) => questions?.length);
+    // @todo (from S23): Remove this eslint ignore and address the missing dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [values.firstChoiceRole, values.secondChoiceRole]);
 
+  if (!questions.length) {
```

Comment:
Removes a conditional react hook call by returning after the hook is called instead of before.